src/test/java/org/jitsi/jibri/util/XmppUtilsKtTest.kt

```diff
@@ -0,0 +1,25 @@
+package org.jitsi.jibri.util
+
+import org.jitsi.jibri.CallUrlInfo
+import org.jxmpp.jid.impl.JidCreate
+import org.testng.annotations.Test
+import kotlin.test.expect
+
+class XmppUtilsKtTest {
+    val baseDomain = "brian.jitsi.net"
+    @Test
+    fun `test basic roomjid url conversion`() {
+        val expected = CallUrlInfo("https://$baseDomain", "roomName")
+        val jid = JidCreate.entityBareFrom("${expected.callName}@$baseDomain")
+
+        expect(expected) { getCallUrlInfoFromJid(jid, "", "brian.jitsi.net") }
+    }
+
+    @Test
+    fun `test roomjid with subdomain url conversion`() {
+        val expected = CallUrlInfo("https://$baseDomain", "roomName")
+        val jid = JidCreate.entityBareFrom("${expected.callName}@mucdomain.$baseDomain")
+
+        expect(expected) { getCallUrlInfoFromJid(jid, "mucdomain.", "brian.jitsi.net") }
+    }
+}
```

Comment:
Didn't you say the kotlin linter yells about not having a new line at the end of the file?

Reply:
i don't think the linter is running on the test files actually.  i don't have it configured to skip them, so maybe it's a default setting. will have to look into it.